### PR TITLE
feat(eslint-plugin): Adds WIP rule for css linting

### PR DESCRIPTION
**Pull request by ling1726**
Adds new rule for css linting. No linting happens in this PR, but all steps until CSS evaluation is done in the rule.

The rule is not exported because it is WIP.

### CODE REVIEW

File: change/@griffel-eslint-plugin-5d860b0c-ef67-4015-a825-eae4b10d1636.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
```
Comment (ling1726) [inline change/@griffel-eslint-plugin-5d860b0c-ef67-4015-a825-eae4b10d1636.json:2]: Staging the rule should not affect public API - the rule and none of its utilties are exported

File: packages/eslint-plugin/src/rules/css-lint/css-lint.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+          throw new Error('No physical filename could be found');
+        }
+
+        const filename = context.getPhysicalFilename();
+        const source = context.getSourceCode().getText();
+        const { metadata } = transformSync(source, {
+          filename,
+          pluginOptions: {
+            generateMetadata: true,
+            ...options,
```
Comment (layershifter) [inline packages/eslint-plugin/src/rules/css-lint/css-lint.ts:119]:
```suggestion
            ...options,
            generateMetadata: true,
```

nit: You probably want to flip this to generate metadata always

File: packages/eslint-plugin/src/rules/css-lint/css-lint.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+          moduleSource: '@griffel/react',
+          importName: 'makeStyles',
+          resetImportName: 'makeResetStyles',
+        },
+        {
+          moduleSource: '@fluentui/react-components',
+          importName: 'makeStyles',
+          resetImportName: 'makeResetStyles',
+        },
+      ],
```
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/eslint-plugin/src/rules/css-lint/css-lint.ts:97]: Maybe I don't understand how to use defaultOptions in eslint - but these should be used if the user does not configure their own options for modules

[the remark this answers is not in the document]
Comment (layershifter) [reply packages/eslint-plugin/src/rules/css-lint/css-lint.ts:97]: Babel plugin has these options as defaults, that's why the question is: do we need them here?

[the remark this answers is not in the document]
Comment (ling1726) [reply packages/eslint-plugin/src/rules/css-lint/css-lint.ts:97]: aaah good catch, got it

File: packages/eslint-plugin/src/rules/css-lint/css-lint.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+type Options = [BabelPluginOptions];
+
+type MessageIds = 'foo';
+
+// TODO createRule does not generate docs url properly for sub-directoryjќ
+export default createRule<Options, MessageIds>({
+  name: 'css-lint',
+  meta: {
+    type: 'problem',
+    schema: [
```
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/eslint-plugin/src/rules/css-lint/css-lint.ts:23]: Will do so in a future PR, when the wiring is a lot clearer

File: packages/eslint-plugin/src/rules/css-lint/css-lint.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/eslint-plugin/src/rules/css-lint/css-lint.ts]: I can make the default option empty, but the eslint types require me to put something there

**Pull request merged**